### ISSUE TITLE
[Feature] generate函数不支持传入inputs_embeds

### ISSUE THREAD

**Issue by ws1993109** [THUDM/ChatGLM2-6B#97]
### Is your feature request related to a problem? Please describe.

当希望在generate时直接传入embedding时，会报错，错误代码如下

opt_tokenss = tokenizer(['你好'], return_tensors="pt",
                         padding="longest",
                         truncation=True,
                         max_length=6,
                         add_special_tokens=False)

model = AutoModel.from_pretrained("./chatglm2-6b", trust_remote_code=True, device='cuda')
model = model.eval()
opt_tokenss.attention_mask = opt_tokenss.attention_mask.to(model.device)
opt_tokenss.input_ids = opt_tokenss.input_ids.to(model.device)
s=time.time()
print(s)
inputs_embeds = model.transformer.embedding.word_embeddings(opt_tokenss.input_ids)
outputs = model.generate(
                input_ids=opt_tokenss.input_ids,
                inputs_embeds=inputs_embeds,
                attention_mask=opt_tokenss.attention_mask,
                do_sample=True,
                top_p=0.8,
                temperature=0.8,
                num_beams=1,
                max_new_tokens=16,
                min_length=1,
                repetition_penalty=1,
                length_penalty=1,
                num_return_sequences=1,
            )
output_text = tokenizer.batch_decode(
                outputs, skip_special_tokens=True
            )
print(output_text)
print('done')

### Solutions

支持在generate时直接传入inputs_embeds，这个PR对GPT2进行了类似的实现：https://github.com/huggingface/transformers/issues/6535#issuecomment-1353658984，chatglm可否进行类似的实现

**Comment by VictorChen2012** [comment]
这个issue有任何更i性能吗？

**Comment by dalong2hongmei** [comment]
解决了吗？也有类似的需要，将embedding部分代码放在用户端，这样客户端可以传输embedding而非明文，有一定程度的信息安全价值
